mypyc/irbuild/builder.py

```diff
@@ -66,6 +67,9 @@
 from mypyc.irbuild.ll_builder import LowLevelIRBuilder
 from mypyc.irbuild.util import is_constant
 
+if TYPE_CHECKING:
+    from mypyc.irbuild.singledispatch import RegisterImplInfo
```

Comment:
I moved `RegisterImplInfo` into mypyc/irbuild/util.py, since that avoids the import cycle and it seemed like a reasonable place to put the type alias definition.